Laboratory/Tests/L4-Standard-Library/Complex.cs

```diff
@@ -0,0 +1,224 @@
+            ("(Complex(0, 1), Complex(0, 1))", "Complex(1, 0)"),
+            ("(Complex(1, 1), Complex(1, 1))", "Complex(1, 0)"),
+            ("(Complex(2, 1), Complex(2, -1))", "Complex(3.div(5), 4.div(5))"),
+            ("(Complex(2, 1), Complex(2, -5))", "Complex(-1.div(29), 12.div(29))"),
+        };
+        [Test]
+        public void ComplexDiv(
+            [ValueSource(nameof(ComplexDivArgsList))] (string lhs, string rhs) args,
+            [Values] EvaluationMode mode
+        )
+        {
+            string complexMul = "_(u:Complex, v:Complex):Complex = u.div(v)";
+            AssertApproxEqual(ValidatedCompilerInput,
+                new FunctionEvaluation(complexMul, args.lhs, mode),
+                new ExpressionEvaluation(args.rhs, mode));
+        }
+        
+        public static (string lhs, string op, string rhs)[] ComplexAddSubArgsList =
+        {
+            ("(Complex(0, 0), Complex(0, 0))", "add", "Complex(0, 0)"),
+            ("(Complex(1, 2), Complex(0, 0))", "add", "Complex(1, 2)"),
+            ("(Complex(4, 7), Complex(1, -5))", "add", "Complex(5, 2)"),
+            ("(Complex(-8, -7.2), Complex(1, 5.2))", "add", "Complex(-7, -2)"),
+            ("(Complex(0, 0), Complex(0, 0))", "sub", "Complex(0, 0)"),
+            ("(Complex(1, 2), Complex(0, 0))", "sub", "Complex(1, 2)"),
+            ("(Complex(4, 7), Complex(1, -5))", "sub", "Complex(3, 12)"),
+            ("(Complex(-8, -7.2), Complex(1, 5.2))", "sub", "Complex(-9, -12.4)"),
+        };
+        [Test]
+        public void ComplexAddSub(
+            [ValueSource(nameof(ComplexAddSubArgsList))] (string lhs, string op, string rhs) args,
+            [Values] EvaluationMode mode
+        )
+        {
+            string complexMul = "_(u:Complex, v:Complex):Complex = u." + args.op + "(v)";
+            AssertApproxEqual(ValidatedCompilerInput,
+                new FunctionEvaluation(complexMul, args.lhs, mode),
+                new ExpressionEvaluation(args.rhs, mode));
+        }
+        
+        
+        public static (string lhs, string rhs)[] ComplexScaleArgsList =
+        {
+            ("(Complex(0, 0), 1)", "Complex(0, 0)"),
+            ("(Complex(13, 91), 0)", "Complex(0, 0)"),
+            ("(Complex(2, 1), 1)", "Complex(2, 1)"),
+            ("(Complex(-3, 4), -2)", "Complex(6, -8)"),
+        };
+        [Test]
+        public void ComplexScale(
+            [ValueSource(nameof(ComplexScaleArgsList))] (string lhs, string rhs) args,
+            [Values] EvaluationMode mode
+        )
+        {
+            string complexScale = "_(c:Complex, n:Num):Complex = c.scale(n)";
+            AssertApproxEqual(ValidatedCompilerInput,
+                new FunctionEvaluation(complexScale, args.lhs, mode),
+                new ExpressionEvaluation(args.rhs, mode));
+        }
+    }
+}
+/**
+MandelbrotIterationState -> iteration:num, z:Complex;
+
+Mandelbrot(x:num, y:num, width:num, height:num, maxIterations:num, maxExtent:num):num
+{
```

Comment:
did you want to commit this commented out code?

Reply:
This was originally on the repository as `Complex.ele` in the same folder. It seemed like it was written as a test for Complex, but not fully implemented. I decided to move it in here and comment it out so we can decide what we want to do with it at another point.

Reply:
okay cool :) sounds good, just wanted to check